docs/guide/custom-functions.md

```diff
@@ -46,25 +71,89 @@ import { FunctionPlugin } from 'hyperformula';
 // start creating a class
 class CountHF extends FunctionPlugin {
 
-// define functions inside this plugin
+  // define functions inside this plugin
   public static implementedFunctions = {
     'HYPER': {
-    // this method's functionality will be defined in the next step
+      // this method's functionality will be defined below
       method: 'hyper',
-    // optionally, mark your function as volatile
+      // set your optional parameters below
+      arrayFunction: false,
+      doesNotNeedArgumentsToBeComputed: false,
+      expandRanges: false,
+      isDependentOnSheetStructureChange: false,
       isVolatile: true,
+      repeatLastArgs: 4,
+      vectorizationForbidden: false,
+    }
+  };
+}
+```
+
+You can set the following optional parameters:
+
+| Option | Type | Description |
+| --- | --- | --- |
+| `arrayFunction` | Boolean | If set to `true`, the function enables the [array arithmetic mode](arrays.md) in its arguments and nested expressions. |
+| `doesNotNeedArgumentsToBeComputed` | Boolean | If set to `true`, the function treats reference or range arguments as arguments that don't create dependency.<br><br>Other arguments are properly evaluated. |
+| `expandRanges` | Boolean | If set to `true`, ranges in the function's arguments are inlined to (possibly multiple) scalar arguments. |
+| `isDependentOnSheetStructureChange` | Boolean | If set to `true`, the function gets recalculated with each sheet shape change (e.g. when adding/removing rows or columns). |
+| `isVolatile` | Boolean | If set to `true`, the function is [volatile](volatile-functions.md). |
+| `repeatLastArgs` | Number | For functions with a variable number of arguments: sets how many last arguments can be repeated indefinitely. |
+| `vectorizationForbidden` | Boolean | If set to `true`, prevents the function from ever being [vectorized](arrays.md#passing-arrays-to-scalar-functions) (but it's up to your function implementation to properly handle vectorization). |
```

Comment:
We should remove this entry as stated by @izulin https://github.com/handsontable/hyperformula/pull/758#discussion_r666980119 that it's internal flag

Reply:
@wojciechczerniak - done: d984703